src/rules/newline-per-chained-call/index.ts

```diff
@@ -0,0 +1,155 @@
+            let hasCallExpression = false;
+
+            if (node.type === 'CallExpression') {
+                hasCallExpression = true;
+            }
+
+            if (
+                (node.parent !== undefined)
+                && node.parent.type !== 'CallExpression'
+                && node.parent.type !== 'MemberExpression'
+            ) {
+                const memberExpressions = [];
+
+                let currentNode = (
+                    node.type === 'CallExpression'
+                        ? node.callee
+                        : node
+                );
+
+                while (
+                    currentNode.type === 'CallExpression'
+                    || currentNode.type === 'MemberExpression'
+                ) {
+                    if (currentNode.type === 'MemberExpression') {
+                        if (
+                            currentNode.property.type === 'Identifier'
+                            && !currentNode.computed
+                        ) {
+                            memberExpressions.push(currentNode);
+                        }
+
+                        currentNode = currentNode.object;
+                    } else if (currentNode.type === 'CallExpression') {
+                        currentNode = currentNode.callee;
+                    }
+                }
+
+                if (
+                    memberExpressions.length > ignoreChainWithDepth
+                    && hasCallExpression
+                    && memberExpressions.some(hasObjectAndPropertyOnSameLine)
+                ) {
+                    memberExpressions
+                        .filter(hasObjectAndPropertyOnSameLine)
+                        .forEach(memberExpression => {
+                            context.report({
+                                node: memberExpression.property,
+                                loc: memberExpression.property.loc.start,
+                                messageId: 'expectedLineBreak',
+                                data: {
+                                    propertyName: getPropertyText(memberExpression),
+                                },
+                                fix: fixer => {
+                                    const firstTokenAfterObject = sourceCode.getTokenAfter(
+                                        memberExpression.object,
+                                        isNotClosingParenToken,
+                                    );
+
+                                    return fixer.insertTextBefore(firstTokenAfterObject!, '\n');
+                                },
+                            });
+                        });
+                }
+            }
+        }
+
+        return {
+            CallExpression: (node: TSESTree.CallExpression): void => {
+                if (node.callee == null || node.callee.type !== 'MemberExpression') {
+                    return;
+                }
+
+                validateCallExpressionIgnoreDepth(node);
```

Comment:
Better move it to inside the branch. It's more readable since you don't have extra cases.